pkg/providers/instancetype/instancetype.go

```diff
@@ -127,6 +133,15 @@ func (p *Provider) LivenessProbe(req *http.Request) error {
 	return nil
 }
 
+func (p *Provider) checkAWSAuth(ctx context.Context) error {
+	_, err := p.ec2api.DescribeInstanceTypesWithContext(ctx, &ec2.DescribeInstanceTypesInput{DryRun: aws.Bool(true)})
+	var aerr awserr.Error
+	if errors.As(err, &aerr) && aerr.Code() == errcode.DryRunOperation {
```

Comment:
Maybe consider creating a `IsDryRunOperationError` and `IgnoreDryRunOperationError` similar to the funcs provided in the client-go errors package.